tianshou/utils/log_tools.py

```diff
@@ -133,15 +168,58 @@ def log_update_data(self, update_result: dict, step: int) -> None:
                 self.write(k, step, v)
             self.last_log_update_step = step
 
+    def save_data(
+        self,
+        epoch: int,
+        env_step: int,
+        gradient_step: int,
+        save_checkpoint_fn: Optional[Callable[[int, int, int], None]] = None,
+    ) -> None:
+        super().save_data(epoch, env_step, gradient_step, save_checkpoint_fn)
+        self.write("save/epoch", epoch, epoch)
+        self.write("save/env_step", env_step, env_step)
+        self.write("save/gradient_step", gradient_step, gradient_step)
+
+    def restore_data(self) -> Tuple[int, float, float, int, int, int, float, int]:
```

Comment:
I think `restore` method should not be implemented in `Logger`, seems quite weird.

Reply:
But if not, where should we put this functionality?